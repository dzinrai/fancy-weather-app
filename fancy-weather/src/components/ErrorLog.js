import React, { useState, useEffect } from 'react';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { useTranslation } from 'react-i18next';

function ErrorLog(props) {
	const { t } = useTranslation();
	let error;
	const [animate, setAnimate] = useState(true);

	if (props.error) error = props.error.statusText;
	if (error === 'Not Found') error = t('Unknown location');
	else if (error === "Browser doesn't support voice enter") error = t("Browser doesn't support voice enter");
	else if (error === "Forbidden") error = t('Background change is not yet available'); 
	else if (error) error = t('Something went wrong'); 
	const scale = () => {
		setAnimate(true);
	};
	useEffect(() => {
		setAnimate(true);
	}, [props.error]);

	if (error) {
		return (
			<div className="error__log">
				<div className={"error-span " + (animate ? 'anim' : '')} onClick={scale} onAnimationEnd={() => setAnimate(false)}>
					<span>{error}</span>
					<FontAwesomeIcon icon='times' onClick={props.clearErrors} className='close__btn' />
				</div>
			</div>
		);
	} else {
		return null;
	}

}

export default ErrorLog;