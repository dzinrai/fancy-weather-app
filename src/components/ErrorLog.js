import React from 'react';

function ErrorLog(props) {
	let error;
	if (props.error) error = props.error.statusText;
	return (
		<div className="error__log">
			{error}
		</div>
	);

}

export default ErrorLog;