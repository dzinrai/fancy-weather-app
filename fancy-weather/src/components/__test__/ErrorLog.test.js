import React from 'react';
import {render, cleanup, screen} from '@testing-library/react';
import renderer from 'react-test-renderer';
import ErrorLog from '../ErrorLog';

afterEach(cleanup);

test('proper error', () => {
    const tree = renderer.create(<ErrorLog error={{statusText: 'Not Found'}} />).toJSON();
    expect(tree).toMatchSnapshot();
})

test('proper error', () => {
    const tree = renderer.create(<ErrorLog error={{statusText: 'Browser doesn\'t support voice enter'}} />).toJSON();
    expect(tree).toMatchSnapshot();
})
