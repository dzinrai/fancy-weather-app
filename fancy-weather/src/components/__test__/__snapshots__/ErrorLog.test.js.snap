// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`proper error 1`] = `
<div
  className="error__log"
>
  <div
    className="error-span anim"
    onAnimationEnd={[Function]}
    onClick={[Function]}
  >
    <span>
      Unknown location
    </span>
  </div>
</div>
`;

exports[`proper error 2`] = `
<div
  className="error__log"
>
  <div
    className="error-span anim"
    onAnimationEnd={[Function]}
    onClick={[Function]}
  >
    <span>
      Browser doesn't support voice enter
    </span>
  </div>
</div>
`;
